app/src/main/java/com/edulectronics/tinycircuit/ui/adapters/CircuitAdapter.java

```diff
@@ -56,12 +57,15 @@ public View getView(int position, View view, ViewGroup parent) {
         LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
         View row = inflater.inflate(R.layout.component, parent, false);
         ImageView image = (ImageView) row.findViewById(R.id.imageView);
-        image.setImageResource(getImageId(position));
-        return row;
-    }
 
-    private int getImageId(int position) {
-        // TODO: replace with components image
-        return R.mipmap.ic_launcher;
+        ImageFactory imageFactory = new ImageFactory();
+        Object object = getItem(position);
+        if(object != null){
+            image.setImageResource(imageFactory.GetImage(object));
```

Comment:
Ik heb er nog even over nagedacht, jouw oplossing is beter denk ik, dus ik zal dat uitvoeren